src/components/Footer/Footer.js

```diff
@@ -1,33 +1,46 @@
 import React, { useState } from 'react';
+import { useLocation, useSearchParams } from 'react-router-dom';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 import { faMap, faListUl } from '@fortawesome/free-solid-svg-icons';
+import DetailFooter from './DetailFooter';
 import FooterDataModal from '../Modal/FooterDataModal';
 import { ETC, RIGHT_ETC } from '../uiData/footerConstData';
 import * as S from './StyleFooter';
 
 const Footer = () => {
-  const [isMap, setIsMap] = useState(false);
   const [isOpenModal, setIsOpenModal] = useState(false);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const location = useLocation();
+  const isMapOpen = searchParams.get('map_open') === 'true';
+
+  if (location.pathname === '/detail' || location.pathname === '/order') {
+    return <DetailFooter />;
+  }
 
   const handleModal = () => {
     setIsOpenModal(prev => !prev);
   };
+
+  const handleNav = value => {
+    searchParams.set('map_open', value);
+    setSearchParams(searchParams);
+  };
+
   return (
     <S.StyleFooter>
-      <S.MapBtn onClick={() => setIsMap(!isMap)}>
-        {isMap ? (
-          <S.MapBtnLink to={`/main/?map_open=${isMap}`}>
-            <S.MapBtnSpan>목록 보기</S.MapBtnSpan>
-            <FontAwesomeIcon icon={faListUl} />
-          </S.MapBtnLink>
-        ) : (
-          <S.MapBtnLink to={`/main/?map_open=${isMap}`}>
-            <S.MapBtnSpan>지도 표시하기</S.MapBtnSpan>
-            <FontAwesomeIcon icon={faMap} />
-          </S.MapBtnLink>
-        )}
-      </S.MapBtn>
-      <S.FooterContainer isMap={isMap}>
+      {isMapOpen ? (
+        <S.MapBtn onClick={() => handleNav(false)}>
+          <S.MapBtnSpan>목록 보기</S.MapBtnSpan>
+          <FontAwesomeIcon icon={faListUl} />
+        </S.MapBtn>
+      ) : (
+        <S.MapBtn onClick={() => handleNav(true)}>
+          <S.MapBtnSpan>지도 보기</S.MapBtnSpan>
+          <FontAwesomeIcon icon={faMap} />
+        </S.MapBtn>
+      )}
```

Comment:
```suggestion
        <S.MapBtn onClick={() => handleNav(!isMapOpen)}>
          <S.MapBtnSpan>{isMapOpen ? "목록 보기" : "지도 보기"}</S.MapBtnSpan>
          <FontAwesomeIcon icon={isMapOpen ? faListUl : faMap} />
        </S.MapBtn>
```
삼항연산자 사용할 때 구조가 변경되는 게 아니라면 이렇게 구현하는것이 더 좋겠네요!